src/itty-router.d.ts

```diff
@@ -29,10 +29,23 @@ export interface Request {
   text?(): Promise<any>
 }
 
-export type Router<TRequest> = {
-  handle: (request: Request, ...extra: any) => any
+export interface IHTTPMethods {
+  get: Route
+  head: Route
+  post: Route
+  put: Route
+  delete: Route
+  connect: Route
+  options: Route
+  trace: Route
+  patch: Route
+}
+
+export type Router<TRequest = Request, TMethods = {}> = {
+  handle: (request: TRequest, ...extra: any) => any
```

Comment:
Could it make sense for this to be `handle: (request: TRequest, ...extra: any) => Promise<any>` instead, given that `handle` is async (ref: https://github.com/kwhitley/itty-router/blob/v2.x/src/itty-router.js#L19)

Reply:
Im for it!  @kwhitley what do you think?  I know you wanted the router to be versatile.  Does that include allowing sync handler function?  As it stands, even though the handler is async, it can still be called sync if the route handlers don't return a promise. If you wanted the handler to be able to be used synchronously we should leave it as return type `any`.  If you want to force handler to be used async we can change it to Promise<any>.

Reply:
It can be called sync (or, just without `await`), but `async` functions will always return a `Promise`, ref. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function